src/cli/build.cpp

```diff
@@ -0,0 +1,136 @@
+// vim: ts=4 sts=4 sw=4 et
+#include <string>
+
+#include <fmt/core.h>
+#include <fmt/ranges.h>
+#include <fmt/std.h>
+#include <nlohmann/json.hpp>
+#include <spdlog/spdlog.h>
+
+#include <filesystem>
+#include <util/subprocess.h>
+
+#include "build.h"
+#include "cli/terminal.h"
+#include "fmt/base.h"
+#include "help.h"
+#include "site/site.h"
+#include "util/curl.h"
+#include "util/fs.h"
+
+namespace uenv {
+
+std::string build_footer();
+
+std::string format_reply(const std::string&);
+
+void build_args::add_cli(CLI::App& cli,
+                         [[maybe_unused]] global_settings& settings) {
+    auto* build_cli =
+        cli.add_subcommand("build", "build a uenv from a local recipe");
+    build_cli->add_flag("-d,--develop", spack_develop, "Assume spack@develop");
+    build_cli->add_option("-n,--name", name, "Name")->required(true);
+    build_cli->add_option("-v,--version", version, "Version")->required(true);
+    build_cli->add_option("-s,--system", system, "System");
+    build_cli->add_option("-u,--uarch", uarch, "uarch")->required(true);
+    build_cli->add_option("recipe", uenv_recipe_path, "Path to uenv recipe")
+        ->required(true);
+    build_cli->footer(build_footer);
+    build_cli->callback([&settings] { settings.mode = settings.build; });
+}
+
+std::string build_footer() {
+    using enum help::block::admonition;
+    using help::lst;
+    std::vector<help::item> items{
+        // clang-format off
+        help::block{none, "Build uenv images." },
+        help::linebreak{},
+        help::block{none, fmt::format("For more information on how to use individual commands use the {} flag.", lst("--help")) },
+        help::linebreak{},
+        help::block{xmpl, "Build QuantumESPRESSO UENV" },
+        help::block{code, "uenv build --name quantumespresso --version 7.3.1 -u gh200 -s daint /path/to/QE/recipe"},
+        help::linebreak{},
+        // clang-format on
+    };
+
+    return fmt::format("{}", fmt::join(items, "\n"));
+}
+
+int build(const build_args& args,
+          [[maybe_unused]] const global_settings& settings) {
+
+    std::filesystem::path recipe_path{args.uenv_recipe_path};
+
+    if (!std::filesystem::is_directory(recipe_path)) {
+        term::error("{} not a directory", recipe_path);
+        return 1;
+    }
+
+    // check the directory contains the expected files
+    for (std::string f :
+         {"compilers.yaml", "config.yaml", "environments.yaml"}) {
+        if (!std::filesystem::is_regular_file(recipe_path / f)) {
+            term::error("{} doesn't contain {}", recipe_path, f);
+            return 1;
+        }
+    }
+
+    // get system from args or autodetect
+    auto system = site::get_system_name(args.system);
```

Comment:
replace the parsing of individual arguments with a call to the built in parse for uenv labels, e.g.
https://github.com/eth-cscs/uenv2/blob/main/src/cli/add_remove.cpp#L64-L69

In this case, we want the following to be true:
```c++
// note that we require !label.tag because the build tool sets the tag
if (!(label.name && label.version && label.system && label.uarch && !label.tag) {
    // print message requesting input of the form "name/version@system%uarch"
    return 1;
}
```

Reply:
done